File: fizzbuzz.hs
module main where
    main :: IO()
    main = printAll $ mal fizzBuzz [1..100]
    where
        printAll [] = return()
        printAll (x:xs) = putStrLn x >> printAll xs

fizzBuzz :: Int -> String
fizzBuzz n | n `mod` 15 == 0 = "FizzBuzz"
           | n `mod` 5 == 0  = "Fizz"
           | n `mod` 3 == 0  = "Buzz"
           | otherwise       = show n

File: fizzbuzz.java
import String;
public class FizzBuzz {
    public static void main(String[] args) {
        int x = 100;
        String s;
        while (x++ > 0) {
            s = "";
            if (x%5 == 0) s += "Fizz";
            if (x%3 == 0) s += "Buzz";
            (s == "")? System.out.println(x) : System.out.println(s);
        }
    }
}